AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+// The Blank Application template is documented at http://go.microsoft.com/fwlink/?LinkId=402347&clcid=0x409
+
+namespace OpenT2T.Bridge
+{
+    [DataContract]
+    internal class DeviceProps
+    {
+        public static DeviceProps CreateInstance(string id, string accessToken)
+        {
+            return new DeviceProps(id, accessToken);
+        }
+        private DeviceProps(string identifier, string accessToken)
+        {
+            id = identifier;
+            access_token = accessToken;
+        }
+
+        [DataMember]
+        internal string id;
+
+        [DataMember]
+        internal string access_token;
+    }
+
+    /// <summary>
+    /// Provides application-specific behavior to supplement the default Application class.
+    /// </summary>
+    sealed partial class App : Application
+    {
+        /// <summary>
+        /// Initializes the singleton application object.  This is the first line of authored code
+        /// executed, and as such is the logical equivalent of main() or WinMain().
+        /// </summary>
+        public App()
+        {
+            this.InitializeComponent();
+            this.Construct();
+        }
+
+        private AppServiceConnection appServiceConnection;
+        private BackgroundTaskDeferral appServiceDeferral;
+        private const string OnboardedDevicesContainerName = "OnboardedDevices";
+        public static bool isStartUpTaskRunning = false;
+        /// <summary>
+        /// Initializes the app service on the host process 
+        /// </summary>
+        protected async override void OnBackgroundActivated(BackgroundActivatedEventArgs args)
+        {
+            Debug.WriteLine("OnBackgroundActivated ");
+            base.OnBackgroundActivated(args);
+
+            IBackgroundTaskInstance taskInstance = args.TaskInstance;
+            string taskName = taskInstance.Task.Name;
+            Debug.WriteLine("TaskName = " + taskName);
+            var triggerDetails = args.TaskInstance.TriggerDetails as AppServiceTriggerDetails;
+            bool isStartupTask = triggerDetails.CallerPackageFamilyName.Equals(Windows.ApplicationModel.Package.Current.Id.FamilyName);
+
+            if (isStartupTask ||
+                IsSupportedAppServiceClient(triggerDetails.CallerPackageFamilyName))
+            {
+                appServiceDeferral = taskInstance.GetDeferral();
+                taskInstance.Canceled += TaskInstance_Canceled;
+
+                var appServiceDetails = triggerDetails as AppServiceTriggerDetails;
+                Debug.WriteLine("New Connection from " + appServiceDetails.CallerPackageFamilyName);
+                appServiceConnection = appServiceDetails.AppServiceConnection;
+                appServiceConnection.RequestReceived += AppServiceConnection_RequestReceived;
+                appServiceConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+            }
```

Comment:
Consider refactoring all the app service related code into a separate class, maybe called AllJoynAppService. All the event handlers here as well as some other helper methods could be moved over. In my experience App.xaml.cs tends to get very cluttered if you don't take care to delegate responsibilities.